File: lib/services/bert_api.dart
import 'dart:convert';
import 'dart:developer' as developer;
import 'package:http/http.dart' as http;

class BertApi {
  // Для локального подключения с реального устройства:
  static const String baseUrl = 'http://192.168.1.100:8000/api';
  // или для эмулятора Android:
  // static const String baseUrl = 'http://10.0.2.2:8000/api';

  // Заглушка метода для генерации предложений из карточек
  Future<Map<String, dynamic>> generateFromCards({
    required List<Map<String, dynamic>> cards,
    String difficulty = 'medium',
    int numSentences = 5,
  }) async {
    try {
      final response = await http.post(
        Uri.parse('$baseUrl/generate-from-cards'),
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode({
          'cards': cards,
          'difficulty': difficulty,
          'num_sentences': numSentences,
        }),
      );

      if (response.statusCode == 200) {
        return jsonDecode(response.body);
      } else {
        final errorData = jsonDecode(response.body);
        throw Exception(errorData['detail'] ?? 'Ошибка при генерации предложений');
      }
    } catch (e) {
      developer.log('Ошибка API: $e', name: 'bert_api');
      // В случае ошибки возвращаем моковые данные
      return {
        'sentences': [
          {
            'masked_text': '我喜欢学习[MASK]语。',
            'original_text': '我喜欢学习中文语。',
            'answers': {'2': '中文'},
            'difficulty': 'medium'
          },
          // Другие моковые предложения...
        ]
      };
    }
  }

  // Проверка здоровья сервера
  Future<bool> checkServerHealth() async {
    try {
      final response = await http.get(Uri.parse('$baseUrl/health'));
      return response.statusCode == 200;
    } catch (e) {
      print('Ошибка при проверке сервера: $e');
      return false;
    }
  }

  // Генерация текста с пропусками
  Future<Map<String, dynamic>> generateBlanks({
    required String text,
    String difficulty = 'medium',
    int numBlanks = 3,
  }) async {
    try {
      final response = await http.post(
        Uri.parse('$baseUrl/generate-blanks'),
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode({
          'text': text,
          'difficulty': difficulty,
          'num_blanks': numBlanks,
        }),
      );

      if (response.statusCode == 200) {
        return jsonDecode(response.body);
      } else {
        final errorData = jsonDecode(response.body);
        throw Exception(errorData['detail'] ?? 'Ошибка при генерации текста с пропусками');
      }
    } catch (e) {
      print('API ошибка (generateBlanks): $e');
      rethrow;
    }
  }

  // Генерация карточек
  Future<Map<String, dynamic>> generateCards({
    String category = 'all',
    String difficulty = 'medium',
    int numCards = 5,
  }) async {
    try {
      final response = await http.post(
        Uri.parse('$baseUrl/generate-cards'),
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode({
          'category': category,
          'difficulty': difficulty,
          'num_cards': numCards,
        }),
      );

      if (response.statusCode == 200) {
        return jsonDecode(response.body);
      } else {
        final errorData = jsonDecode(response.body);
        throw Exception(errorData['detail'] ?? 'Ошибка при генерации карточек');
      }
    } catch (e) {
      print('API ошибка (generateCards): $e');
      rethrow;
    }
  }
} 

File: mobile_app/api/bertApi.js
/**
 * API клиент для взаимодействия с BERT-сервером
 */
const BASE_URL = 'http://your-bert-server-address:8000/api';

/**
 * Генерирует текст с пропусками на основе предоставленного текста
 * @param {string} text - Исходный текст на китайском
 * @param {string} difficulty - Сложность: 'easy', 'medium', 'hard'
 * @param {number} numBlanks - Количество пропусков
 * @returns {Promise} - Объект с текстом с пропусками и ответами
 */
export const generateBlanks = async (text, difficulty = 'medium', numBlanks = 3) => {
  try {
    const response = await fetch(`${BASE_URL}/generate-blanks`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        text,
        difficulty,
        num_blanks: numBlanks,
      }),
    });

    if (!response.ok) {
      const errorData = await response.json();
      throw new Error(errorData.detail || 'Ошибка при генерации текста с пропусками');
    }

    return await response.json();
  } catch (error) {
    console.error('API ошибка (generateBlanks):', error);
    throw error;
  }
};

/**
 * Получает случайные карточки с текстами и пропусками
 * @param {string} category - Категория текстов ('greeting', 'food', 'travel', 'all')
 * @param {string} difficulty - Сложность: 'easy', 'medium', 'hard'
 * @param {number} numCards - Количество карточек
 * @returns {Promise} - Массив карточек с текстами и пропусками
 */
export const generateCards = async (category = 'all', difficulty = 'medium', numCards = 5) => {
  try {
    const response = await fetch(`${BASE_URL}/generate-cards`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        category,
        difficulty,
        num_cards: numCards,
      }),
    });

    if (!response.ok) {
      const errorData = await response.json();
      throw new Error(errorData.detail || 'Ошибка при генерации карточек');
    }

    return await response.json();
  } catch (error) {
    console.error('API ошибка (generateCards):', error);
    throw error;
  }
};

/**
 * Проверяет доступность сервера
 * @returns {Promise<boolean>} - true, если сервер доступен
 */
export const checkServerHealth = async () => {
  try {
    const response = await fetch(`${BASE_URL}/health`);
    return response.ok;
  } catch (error) {
    console.error('API ошибка (checkServerHealth):', error);
    return false;
  }
}; 